mkosi/__init__.py

```diff
@@ -3339,6 +3339,14 @@ def make_image(
 
     logging.debug(json.dumps(output, indent=4))
 
+    if context.config.verity == Verity.defer and ArtifactOutput.roothash in context.config.split_artifacts:
+        for k in output:
+            # When the partition is deferred/skipped the roothash is listed as the literal 'TDB' string
+            if k["label"].endswith("_verity") and k["roothash"] != "TBD":
+                (context.staging / f"{context.config.output_with_format[:-3]}roothash").write_text(
+                    k["roothash"]
+                )
```

Comment:
This is the wrong place to put this logic, instead, just modify the two callsites of `finalize_roothash()` to write the roothash to the split file if requested.

Reply:
But those only run when kernel/uki are built? This is needed every time repart runs and builds a dm-verity image, which can be done independently and without ukis

Reply:
Ah in that case please beef up `finalize_roothash()` and use it here.

Reply:
done